lib/pusher/webhook.rb

```diff
@@ -31,10 +31,10 @@ class WebHook
     def initialize(request, client = Pusher)
       @client = client
       # For Rack::Request and ActionDispatch::Request
-      if request.respond_to?(:env) && request.respond_to?(:content_type)
+      if request.respond_to?(:env) && (request.methods & [:media_type, :content_type]).any?
         @key = request.env['HTTP_X_PUSHER_KEY']
         @signature = request.env["HTTP_X_PUSHER_SIGNATURE"]
-        @content_type = request.content_type
+        @content_type = request.media_type || request.content_type
```

Comment:
@sonologico Just to be sure, do you propose to remove any occurrence of `content_type` in this snippet and just read  directly the `request.media_type`?

Reply:
Yes. This is what I meant.